### DISCUSSION TITLE
posts/glibc-static-linking/

### DISCUSSION THREAD

**Discussion by giscus** [KonishchevDmitry/blog#6] [Announcements]
# posts/glibc-static-linking/

В своей работе я не раз сталкивался с тем, что, собрав Go/Rust-программу на своей рабочей машине и скопировав её на другую, с более старой версией дистрибутива, есть большой шанс того, что она при запуске упадёт с ошибкой вида /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.38' not found. Обычно разбираться всегда было некогда, и я просто пересобирал свою программу на нужной версии дистрибутива, но вот тут стало интересно – и я пошёл посмотреть, как скомпилить свою Rust’овую программу статически с glibc, чтобы не иметь таких проблем.

https://konishchev.ru/posts/glibc-static-linking/

**Comment by ikar49** [comment]
Хочется отметить, что в данной опусе нет ничего про проблемы именно обратной совместимости GLIBC. Только проблемы двойственной загрузки одной и той же библиотеки в двух версиях в одно адресное пространство.
Однако данная статья захламляет выборку интернет-поисковиков и отравляет ИИ-шные рецензии.
